schemas/component/ui.schema

```diff
@@ -0,0 +1,113 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/ui.schema",
+  "title": "UI Schema",
+  "description": "Visual customization options for BotFramework SDK components.",
+  "type": "object",
+  "patternProperties": {
+      "^\\$schema$": {
+          "type": "string"
+      },
+      "^(?!\\$role:).+$": {
+          "title": "$kind",
+          "description": "A map of component names to UI options with customizations for each component.",
+          "$ref": "#/definitions/UIOptionsRecursive"
+      },
+      "^\\$role:.+$": {
```

Comment:
This may not be necessary to expose at the SDK level and may be more of a Composer only customization. We use this mechanism to provide some custom UI for expressions.

Reply:
Seems like I can achieve the same effect on an individual property.  This is a "global" default.  Today we do not allow $roles that we did not define so this really boils down to the question of allowing someone to customize all properties with a $role.  I'm not sure if that is worth it or not.  

---
In reply to: [451157108](https://github.com/microsoft/botframework-sdk/pull/5955#discussion_r451157108) [](ancestors = 451157108)

Reply:
I think I am going to remove this and only expose it through the JS interface for the moment, if at all.